src/main/java/org/jenkinsci/main/modules/sshd/IdleTimeout.java

```diff
@@ -42,13 +44,13 @@ public void apply(SshServer sshd) {
 			return;
 		}
 
-		sshd.getProperties().put(ServerFactoryManager.IDLE_TIMEOUT, timeoutInMilliseconds);
+		CoreModuleProperties.IDLE_TIMEOUT.set(sshd, Duration.ofMillis(timeoutInMilliseconds));
 		// Read timeout must also be changed
 		long readTimeout = 0;
 		if (timeoutInMilliseconds != 0) {
-			readTimeout = ServerFactoryManager.DEFAULT_NIO2_READ_TIMEOUT - ServerFactoryManager.DEFAULT_IDLE_TIMEOUT + timeoutInMilliseconds;
+			readTimeout = ADDITIONAL_TIME_FOR_TIMEOUT + timeoutInMilliseconds;
```

Comment:
`DEFAULT_NIO2_READ_TIMEOUT` now is 0 so the old logic has the opposite behavior as expected, due to we have to set `DEFAULT_NIO2_READ_TIMEOUT` higher than `IDLE_TIMEOUT`, I have set a constant of one minute that increases the `DEFAULT_NIO2_READ_TIMEOUT`.